CommonTools/PileupAlgos/src/PuppiContainer.cc

```diff
@@ -150,66 +140,69 @@ void PuppiContainer::getRMSAvg(int iOpt,
     bool pCharged = fPuppiAlgo[pPupId].isCharged(iOpt);
     double pCone = fPuppiAlgo[pPupId].coneSize(iOpt);
     //Compute the Puppi Metric
-    if (!pCharged)
-      pVal = goodVar(iConstits[i0], iParticles, pAlgo, pCone);
-    if (pCharged)
-      pVal = goodVar(iConstits[i0], iChargedParticles, pAlgo, pCone);
+    double pVal = -1;
+    // calculate goodVar only for candidates that (1) will not be assigned a predefined weight (e.g 0, 1),
+    // or (2) are required for computations inside puppi-algos (see call to PuppiAlgo::add below)
+    if (((not(fApplyCHS and ((iConstits[i0].id == 1) or (iConstits[i0].id == 2)))) and (iConstits[i0].id != 3)) or
+        ((std::abs(iConstits[i0].eta) < fPuppiAlgo[pPupId].etaMaxExtrap()) and
+         ((iConstits[i0].id == 1) or (iConstits[i0].id == 2)))) {
```

Comment:
```suggestion
    bool pvRelated = iConstits[i0].id == 1 or iConstits[i0].id == 2;
    if (not(fApplyCHS and pvRelated) or
        (pvRelated and std::abs(iConstits[i0].eta) < fPuppiAlgo[pPupId].etaMaxExtrap()) ) {
```
(assuming I didn't get lost in parentheses) 
- IIUC, iConstits == 3 were already removed, so it's always true
- a common piece is precomputed
- somewhat excessive use of parentheses is reduced